Mark headings in XLIFFs so that ATE can indicate that these fields have SEO significance.

kagg-design opened the pull request:
Changes in wpml-page-builders-elementor to support task of the ticket.

https://onthegosystems.myjetbrains.com/youtrack/issue/wpmltm-3081

Igor-WPML committed:
Mark headings in XLIFFs so that ATE can indicate that these fields have SEO significance.

https://onthegosystems.myjetbrains.com/youtrack/issue/wpmltm-3081

brucepearson commented on src/class-wpml-elementor-translatable-nodes.php:
```diff
@@ -152,6 +152,13 @@ private function get_integration_classes( $node_data ) {
+		if ( 'heading' === $settings[ $this->type ] ) {
+			$header_size = isset( $settings[ $this->settings_field ]['header_size'] ) ? $settings[ $this->settings_field ]['header_size'] : '';
+			if ( $header_size ) {
+				return $field['field'] . '-' . $settings[ $this->type ] . '-' . $node_id . '-' . $header_size;
```
@Igor-WPML I'm not sure this is the best approach. The name of the string will change if the user changes the heading size and I expect they will then loose any existing translations they had.

kagg-design replied:
@brucepearson I checked this carefully, and yes Bruce, you are right (( Working on solving the issue.

kagg-design replied:
@brucepearson  Following to Pierre's advice, I have added column to strings table. This solves the issue. Please also refer to https://github.com/OnTheGoSystems/wpml-page-builders/pull/63

Igor-WPML committed:
Get 'wrap' field from registered string.

https://onthegosystems.myjetbrains.com/youtrack/issue/wpmltm-3081

Igor-WPML committed:
Rename method get_string_wrap() to get_wrap_tag() for consistency.

Add test for this method.

https://onthegosystems.myjetbrains.com/youtrack/issue/wpmltm-3081

strategio reviewed (dismissed):
+1 @kagg-design

brucepearson commented on src/class-wpml-elementor-translatable-nodes.php:
```diff
@@ -155,6 +157,25 @@ public function get_string_name( $node_id, $field, $settings ) {
+	 */
+	private function get_wrap_tag( $settings ) {
+		if ( isset( $settings[ $this->type ] ) && 'heading' === $settings[ $this->type ] ) {
+			// h2 must be by default, as Elementor sends empty header size for h2 block.
```
@kagg-design You could replace this comment with a named constant.

```
const DEFAULT_HEADING_TAG = 'h2';

...
...
$header_size = isset( $settings[ $this->settings_field ]['header_size'] ) ? $settings[ $this->settings_field ]['header_size'] : self::DEFAULT_HEADING_TAG;
```

kagg-design replied:
@brucepearson done

brucepearson commented on tests/phpunit/tests/test-wpml-elementor-integration-factory.php:
```diff
@@ -26,6 +26,7 @@ public function it_creates() {
 			                     'WPML_Elementor_Adjust_Global_Widget_ID_Factory',
 			                     'WPML_PB_Elementor_Handle_Custom_Fields_Factory',
 			                     'WPML_Elementor_Media_Hooks_Factory',
+			                     'WPML_Elementor_WooCommerce_Hooks_Factory',
```
@kagg-design What is this for?

kagg-design replied:
@brucepearson it is the fix of previous bug. Test failed without it even before my MR.

Igor-WPML committed:
Use named constant for default heading.

https://onthegosystems.myjetbrains.com/youtrack/issue/wpmltm-3081

andreasciamanna commented on src/class-wpml-elementor-translatable-nodes.php:
```diff
@@ -6,7 +6,10 @@
+
+	const TYPE = 'widgetType';
+
+	const DEFAULT_HEADING_TAG = 'h2';
```
@Igor-WPML we can save some bytes not using empty lines between declarations of the same type:

```php
const SETTINGS_FIELD      = 'settings';
const TYPE                = 'widgetType';
const DEFAULT_HEADING_TAG = 'h2';
```

kagg-design replied:
@andreasciamanna I proceed in this way, because otherwise Ctrl+Alt+L (Reformat Code) in phpStorm formats lines like that

    const SETTINGS_FIELD = 'settings';
    const TYPE = 'widgetType';
    const DEFAULT_HEADING_TAG = 'h2';

and immediately reports

    phpcs: Generic.Formatting.MultipleStatementAlignment.NotSameWarning: Equals sign not aligned with surrounding assignments; expected 16 spaces but found 1 space

andreasciamanna replied:
@kagg-design just fix your IDE settings then:

![image](https://user-images.githubusercontent.com/181780/53243941-b379fd00-36a9-11e9-8207-8fab96b7d148.png)

![image](https://user-images.githubusercontent.com/181780/53243930-ac52ef00-36a9-11e9-98c4-1d4b47167b2e.png)

kagg-design replied:
@andreasciamanna thank you for explanations. I completely forgot about these settings.

Constants now are aligned as proposed.

andreasciamanna reviewed (dismissed):
I don't have a full understanding of what's going on here, but it seems alright to me.

Igor-WPML committed:
Format constants properly.

https://onthegosystems.myjetbrains.com/youtrack/issue/wpmltm-3081
